app/src/main/java/com/coveros/coverosmobileapp/blogpost/CommentFormActivity.java

```diff
@@ -36,24 +44,72 @@ public void onClick(View v) {
                 email = enterEmail.getText().toString();
                 message = enterMessage.getText().toString();
 
+                List<String> emptyFields = checkFieldIsEmpty(author, email, message);
+                if (!emptyFields.isEmpty()) {
+                    emptyFieldAlertDialog = createEmptyFieldAlertDialog(emptyFields);
+                    emptyFieldAlertDialog.show();
+                }
+
                 // logging for now until we make the actual request
                 Log.d("AUTHOR", author);
                 Log.d("EMAIL", email);
                 Log.d("MESSAGE", message);
             }
         });
+
     }
 
-    public String getAuthor() {
+    static List<String> checkFieldIsEmpty(String author, String email, String message) {
+        List<String> emptyFields = new ArrayList<>();
+        if (author.isEmpty()) {
+            emptyFields.add("name");
+        }
+        if (email.isEmpty()){
+            emptyFields.add("email");
+        }
+        if (message.isEmpty()) {
+            emptyFields.add("message");
+        }
+        return emptyFields;
+    }
+
+    private AlertDialog createEmptyFieldAlertDialog(List<String> emptyFields) {
+        AlertDialog emptyFieldAlertDialog = new AlertDialog.Builder(CommentFormActivity.this).create();
+        emptyFieldAlertDialog.setTitle(R.string.empty_field_alert_dialog_title);
+        String emptyFieldsString;
+        if (emptyFields.size() == 1) {
```

Comment:
in Java 8 this would be a good case for Stream.reduce - since we're in Java 7 this is good, but thought I'd mention it